climakitae/data_export.py

```diff
@@ -0,0 +1,225 @@
+        if ('simulation' in data_to_export.coords): # if simulation coord exists
+            if ('simulation' not in kwargs): # and one simulation is not supplied by user
+                if (np.size(data_to_export.coords['simulation'].values) > 1): # and there is > 1
+                    raise ValueError("File format does not allow for data from"+
+                            " more than one simulation. Please pass simulation = 'sim'"+
+                             " anywhere after file_name in the export_dataset() call."+
+                             " E.g., 'app.export_dataset(data,file_name='example',"+
+                            "simulation='cesm2')")
+                else: # automatically pull the simulation name if there is 1
+                    simu = data_to_export.coords['simulation'].values                    
+            else: # it is a kwarg
+                assert type(kwargs['simulation']) is str,("Please pass simulation name "+
+                "in double or single quotation marks.")
+                simu = kwargs['simulation']
+                    
+            sim_dict = {'simulation_model' : simu}
+            # ds_attrs.update(sim_dict)
+            data_to_export['simulation'].attrs = {'simulation_model' : simu}
+            data_to_export = data_to_export.sel(simulation=simu) # subset        
+    
+    # same as above, but for scenario instead of simulation
+        if ('scenario' in data_to_export.coords): 
+            if ('scenario' not in kwargs): 
+                if (np.size(data_to_export.coords['scenario'].values) > 1):
+                    raise ValueError("File format does not allow for data from"+
+                            " more than one scenario. Please pass scenario = 'scenario'"+
+                             " anywhere after file_name in the export_dataset() call."+
+                             " E.g., 'app.export_dataset(data,file_name='example',"+
+                            "scenario='historical')")
+                else: 
+                    scen = data_to_export.coords['scenario'].values                    
+            else: 
+                assert type(kwargs['scenario']) is str,("Please pass scenario name "+
+                "in double or single quotation marks.")
+                scen = kwargs['scenario']
+                    
+            scen_dict = {'climate_scenario' : scen}
+            # ds_attrs.update(scen_dict)
+            data_to_export['scenario'].attrs = scen_dict
+            data_to_export = data_to_export.sel(scenario=scen) # subset                    
+   
+    data_to_export.attrs = ds_attrs
+    data_to_export = data_to_export.squeeze(drop=True)
+    file_size_threshold = 5 # in GB
+    bytes_per_gigabyte = 1024*1024*1024    
+    disk_space = shutil.disk_usage('./')[2]/bytes_per_gigabyte
+    data_size = data_to_export.nbytes/bytes_per_gigabyte
+    
+    if (disk_space < data_size):
+        raise ValueError("Not enough disk space to export data!"+
```

Comment:
Is this something we want user feedback on?  Will 10GB be enough memory?

Reply:
it's disk space, not memory. And I don't know if we can change that - it depends on resources.